activesupport/CHANGELOG.md

```diff
@@ -1,3 +1,28 @@
+*   Add expiry and purpose to MessageEncryptor & MessageVerifier.
+    
+    Messages encrypted and generated using MessageEncryptor and Message Verifier now
+    ActiveSupport expiry and purpose. You can set the purpose of the message using the key
+    :purpose, and likewise, the expiration using :expires_at or :expires_in.
+
+    This ensures that the message is used only for its intended purpose and also while it
+    hasn't expired.
+
+    Usage example: Message Encryptor
+
+        token = @encryptor.encrypt_and_sign(address, purpose: "shipping", expires_at: Time.now.end_of_year)
+        @encryptor.decrypt_and_verify(token, purpose: "shipping")
+
+    Usage example: Message Verifier
+
+        token = @verifier.generate(address, purpose: "shipping", expires_in: 1.month)
+        @verifier.verified(token, purpose: "shipping")
+        @verifier.verify(token, purpose: :shipping)
```

Comment:
Think this is too detailed for a changelog entry. How does this attempt at condensing the description sound?

```
For instance, to ensure a message is only usable for one intended purpose:

    token = @verifier.generate("x", purpose: :shipping)

    @verifier.verified(token, purpose: :shipping) # => "x"
    @verifier.verified(token)                     # => nil

Or make it expire after a set time:

    @verifier.generate("x", expires_in: 1.month)
    @verifier.generate("y", expires_at: Time.now.end_of_year)

Showcased with `ActiveSupport::MessageVerifier`, but works the same for 
`ActiveSupport::MessageEncryptor`'s `encrypt_and_sign` and `decrypt_and_verify`.
```